### ISSUE TITLE
`requestPermission` should not call explicitly `setSwipeLock`

### ISSUE THREAD

**Issue by cortinico** [AppIntro/AppIntro#905]
I believe we should cleanup the code around `requestPermission` to don't call explicitly `setSwipeLock(true)` and `setSwipeLock(false)` but instead operate on the `isFullPagingEnabled` and potentially retain the value set by the user.

> Knowing that, I noticed in your library's code that `AppIntroBase.onRequestPermissionsResult` makes a hard call to `setSwipeLock(false)`. Isn't that call interfering with my own ? Should I make my call in another override than `onSlideChanged` to avoid that ?

Yup, that's exactly the case. If you call `setSwipeLock(false)` inside the `onRequestPermissionsResult` (provided that you call also `super.onRequestPermissionsResult` before) it should work properly 👍

_Originally posted by @cortinico in https://github.com/AppIntro/AppIntro/issues/902#issuecomment-727539114_

**Comment by DeonWaju** [comment]
I am new here, Looking for an issue to pick up, can i pick this?

**Comment by cortinico** [comment]
Yup you can pick this up @DeonWaju but it might not be the easiest for a new joiner

**Comment by DeonWaju** [comment]
Would you mind giving me something someone new would do? @cortinico

**Comment by cortinico** [comment]
I don't have anything on top of my mind. @paolorotolo maybe you have some ideas?

**Comment by paolorotolo** [comment]
Hey @DeonWaju, it's really awesome you want to contribute! I don't think we have issues for starters right now, but you can help test the `release/6.x` branch and report/fix new issues with viewpager2.

Anyway, we're going off-topic here: if you want to have a chat join `#appintro` channel in [KotlinLang Slack](https://slack-chats.kotlinlang.org/?_gl=1*1pytj2p*_ga*ODA5MTI4NDI5LjE2ODExMzU2ODI.*_ga_9J976DJZ68*MTcwMDg3MDgxNS4zOS4wLjE3MDA4NzA4MTUuNjAuMC4w&_ga=2.257961929.212420555.1700870816-809128429.1681135682).
